tests/python/test_ndarray.py

```diff
@@ -607,3 +616,15 @@ def init(d: ti.i32, arr: ti.types.ndarray()):
     y = ti.ndarray(dtype=ti.f32, shape=(n2, n2))
     init(3, y)
     assert (y.to_numpy() == (np.ones(shape=(n2, n2)) * 3)).all()
+
+
+@test_utils.test(arch=supported_archs_taichi_ndarray)
+def test_generation():
+    curr_arch = ti.lang.impl.current_cfg().arch
+    n1 = 4
+    x = ti.ndarray(dtype=ti.f32, shape=(n1, n1))
+    curr_gen = x._gen
+    ti.reset()  # gen++
+    ti.init(curr_arch)  # calls ti.reset(), gen++
+    y = ti.ndarray(dtype=ti.f32, shape=(n1, ))
+    assert y._gen == curr_gen + 2
```

Comment:
Super nit, we probably don't care the specific difference between these two gens, so it would be fine to just `assert y._gen != curr_gen`  (O.w. if we change the generation implementation, this test would unexpectedly fail).

Also, would be nice to call a `del x` in the end :-)